### PR TITLE
dev: clean send_raw_transaction

### PR DESCRIPTION
**Pull request by greged93**
Time spent on this PR: 0.1 day

Resolves: #NA

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
Clean the layout of the `send_raw_transaction` in order to make it more readable.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/bin/hive_chain.rs [hunk 1]
```diff
@@ -67,10 +67,15 @@ async fn main() -> eyre::Result<()> {
             let chain_id = transaction.chain_id().ok_or(eyre!("Failed to recover chain id"))?;
             let starknet_tx = to_starknet_transaction(&transaction, chain_id, signer, u64::MAX)?;
 
+            let nonce = match &starknet_tx {
```
Comment (Eikix) [inline src/bin/hive_chain.rs:70]: I think starknet tx has a getter for nonce

File: src/eth_provider/provider.rs [hunk 2]
```diff
@@ -859,3 +798,50 @@ where
         }
     }
 }
+
+#[cfg(feature = "hive")]
+impl<SP> EthDataProvider<SP>
+where
+    SP: starknet::providers::Provider + Send + Sync,
+{
+    async fn deploy_evm_transaction_signer(&self, signer: Address) -> EthProviderResult<()> {
```
Comment (Eikix) [inline src/eth_provider/provider.rs:807]: nice! i think this deserves a bit of context in the form of rustdoc

File: src/bin/hive_chain.rs (hunk 1 above)
Comment (greged93) [reply src/bin/hive_chain.rs:70]: Unfortunatly not :/

**Review by Eikix — APPROVED** [review]
lgtm

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**